pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -291,6 +293,20 @@ func managePod(configMapsGetter corev1client.ConfigMapsGetter, secretsGetter cor
 		required.Spec.Containers[0].Args = append(required.Spec.Containers[0].Args, "--tls-private-key-file=/etc/kubernetes/static-pod-resources/secrets/serving-cert/tls.key")
```

Comment:
`manageKubeControllerManagerConfig()` needs to use `MergePrunedConfigMap()` instead, this function removes fields that are not part of the config object from `observedConfig`